### PR TITLE
chore(client): ts migration

### PR DESCRIPTION
**Pull request by jmetev1**
Checklist:

- [ ] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [ ] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [ ] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/templates/Challenges/utils/worker-executor.ts [hunk 1]
```diff
@@ -26,12 +40,13 @@ class WorkerExecutor {
         if (e.data?.type === 'contentLoaded') {
           resolve(newWorker);
         }
+        // do something else if not contentLoaded?
```
Comment (jmetev1) [inline client/src/templates/Challenges/utils/worker-executor.ts:43]: i'm in search of some advice. getworker is supposed to return a worker by calling createworker. but if createworker doesn’t get ‘contentloaded’ or has an error, then it won’t return a worker. Right now the failure to get ‘contentloaded’ would result in the call to getWorker inside execute never resolving and the whole thing getting stuck. Typescript is forcing me to address this more directly and so I’ve gone down this path but I wanted some feedback if it seems reasonable?

Comment (jmetev1) [reply client/src/templates/Challenges/utils/worker-executor.ts:43]: You can see it get stuck by changing the test. If the mockworker doesn't call 'contentloaded', the whole thing will hang till jest shuts it down.

Comment (jmetev1) [reply client/src/templates/Challenges/utils/worker-executor.ts:43]: i know @ojeytonwilliams  had worked on this code before. is it cool to tag you for advice here?

**Comment by raisedadead** [comment]
Consider opening PRs as draft (not the same as "convert to draft" after the fact) if you have something in the works. It helps cut the notification noise.

I appreciate your understanding.

File: client/src/templates/Challenges/utils/worker-executor.ts (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/templates/Challenges/utils/worker-executor.ts:43]: I'm afraid I've never understood this code particularly well, so I can't give particularly useful advice.

All I can say is that, despite being confusing, this code works well. So, if you need to use `@ts-expect-error` to keep TS from complaining, that's totally fine.

**Comment by jmetev1** [comment]
> Consider opening PRs as draft (not the same as "convert to draft" after the fact) if you have something in the works. It helps cut the notification noise.
> 
> I appreciate your understanding.

Actually I never knew that was an option. Will do.

**Comment by Sembauke** [comment]
Hey @jmetev1,

I will be closing this pull-request as it has been abandoned. Feel free to revisit/re-open when you feel like it.

**Pull request closed without merging**